src/design-system/components/Checkbox.tsx

```diff
@@ -52,66 +52,62 @@ export const Checkbox = forwardRef<View, CheckboxProps>(
       return 'fg.white'
     }, [disabled, value])
 
-    const bgColor = useMemo<ColorNames>(() => {
-      if (!value) {
-        return 'fg.white'
-      }
+    const bgColor = useMemo<ColorNames | undefined>(() => {
       if (disabled) {
         return 'bg.disabled_subtle'
       }
 
+      if (!isChecked) {
+        return undefined
+      }
+
       return 'bg.brand.solid'
-    }, [disabled, value])
+    }, [disabled, isChecked])
 
-    const borderColor = useMemo<ColorNames>(
-      () =>
-        disabled
-          ? 'border.disabled'
-          : isError
-          ? 'border.error'
-          : value
-          ? 'bg.brand.solid'
-          : 'border.primary',
-      [value, isError, disabled]
-    )
+    const borderColor = useMemo<ColorNames | undefined>(() => {
+      if (disabled) {
+        return 'border.disabled'
+      }
+      if (isError) {
+        return 'border.error'
+      }
+
+      if (isChecked) {
+        return 'bg.brand.solid'
+      }
+
+      return 'border.primary'
+    }, [isChecked, isError, disabled])
 
     return (
-      <Pressable
+      <TouchableOpacity
         {...{ disabled, hitSlop, ref }}
         onPress={handleValueChange}
         style={styles.mainContainer}
       >
         <View style={styles.row}>
```

Comment:
This View, probably could be removed, as in TouchableOpacity there are no other children